src/CommonProviderImplementation/Helpers.fs

```diff
@@ -147,43 +148,18 @@ module ProviderHelpers =
           SampleIsWebUri : bool
           SampleIsResource : bool }
 
-    type private EmbeddedResourceReader() =
```

Comment:
If we're reading resources via AssemblyReader, then that is using an in-memory snapshot of the bytes of the target assembly, and so there should be no file locking.

(Mind you, that's also chewing through a non-trivial amount of memory, which needs to be solved, I'm trying to work out what we can do about that - if anything)